dailystats.py

```diff
@@ -30,31 +33,21 @@ def read_stats_file(csv_file_name):
     :return: A dictionary mapping countries/regions to a list of dates and daily
         case counts.
     """

```

Comment:
Here I've used the standard `csv` library, which removes a lot of the low level operations.

`with open` makes sure the file is closed even on a failures, and `defaultdict` is a dict which will generate a default for keys. In this case, any key defaults to an empty list, so you can always do `result[key].append(value)`

In your current implementation there is a bug as well - on line 48 it should be
```python
            the_dictionary[field_list[0]] = [new_line_value_list]
```

which is why you had to skip the first item in subsequent code